src/server/cmd/pachd/main.go

```diff
@@ -408,19 +413,27 @@ func getEtcdClient(etcdAddress string) discovery.Client {
 
 const clusterIDKey = "cluster-id"
 
-func getClusterID(client discovery.Client) (string, error) {
-	id, err := client.Get(clusterIDKey)
+func getClusterID(client *etcd.Client) (string, error) {
+	resp, err := client.Get(context.Background(),
+		clusterIDKey)
+
 	// if it's a key not found error then we create the key
-	if err != nil && strings.Contains(err.Error(), "not found") {
+	if resp.Count == 0 {
 		// This might error if it races with another pachd trying to set the
 		// cluster id so we ignore the error.
-		client.Set(clusterIDKey, uuid.NewWithoutDashes(), 0)
+		client.Put(context.Background(), clusterIDKey, uuid.NewWithoutDashes())
 	} else if err != nil {
 		return "", err
 	} else {
+		if resp.Count != 1 {
+			return "", fmt.Errorf("got an unexpected number of PPS tokens: %d", resp.Count)
+		}
```

Comment:
This makes me uneasy, if we ever do somehow get into this situation won't that mean that the cluster is permanently unusable because calls the `getClusterID` are failing?

Reply:
That is true. I'm not sure what a good default here would be. Read the first value only and print a warning?

Reply:
Actually I think you can just ignore this case, since I don't think a vanilla `Get` like this can ever return more than 1 Kv. I'd just return `resp.Kvs[0].Value` and assume it's the only one without printing a warning. I think it's very unlikely that warning will ever get triggered but it's likely to confuse us later when we're reading this code and wondering in what circumstances that could ever come up. A comment documenting this could be nice though.